sympy/polys/polytools.py

```diff
@@ -7938,3 +7937,248 @@ def named_poly(n, f, K, name, x, polys):
+        raise DomainError("All polynomials must have the same domain and generators")
+
+    constant_conditions = []
+    non_constant_polys = []
+
+    for poly in polys:
+        if poly.total_degree() == 0:
+            if poly.is_zero:
+                continue
+            elif poly.as_expr().is_number:
+                if not poly.is_zero:
+                    return [], None
+            else:
+                constant_conditions.append(poly)
+        else:
+            non_constant_polys.append(poly)
+
+    if not non_constant_polys:
+        if constant_conditions:
+            return [], constant_conditions
+        return [[]], None
+
+    factored_eqs = []
+    for poly in non_constant_polys:
+        constant, factors = poly.factor_list()
+        current_factors = []
+
+        if constant.is_zero and not factors:
+            continue
+
+        if factors:
+            if constant.is_number and constant != 0:
+                current_factors.extend((factor, []) for factor, _ in factors)
+            else:
+                constant_poly = Poly(constant, *base_gens, domain=base_domain)
+                current_factors.extend((factor, [constant_poly]) for factor, _ in factors)
+        elif not constant.is_zero:
+            return [], None
+
+        if current_factors:
+            factored_eqs.append(current_factors)
+
+    if not factored_eqs:
+        return [[]], constant_conditions if constant_conditions else None
+
+    return _combine_factor_systems_poly(factored_eqs), constant_conditions
+
+
+def factor_system_cond(eqs, gens=None, **kwargs):
+
+    polys, result = _process_system_equations(eqs, gens, **kwargs)
+    if result is not None:
+        solutions, conditions = result
+        if conditions is None:
+            if not solutions:
+                return [], None
+            else:
+                return [[]], None
+        if conditions:
+            return solutions, [Eq(cond, 0) for cond in conditions]
+        return solutions, None
+
+    poly_results, constant_conditions = factor_system_poly(polys, **kwargs)
+
+    if not poly_results and constant_conditions is None:
+        return [], None
+
+    results = []
+    for combo in poly_results:
+        factor_conditions = {}
+        for factor, conditions in combo:
+            expr = factor.as_expr()
+            if expr not in factor_conditions:
+                factor_conditions[expr] = []
+            factor_conditions[expr].extend(conditions)
+
+        boolean_combo = []
+        for factor, conditions in factor_conditions.items():
+            if not conditions:
+                boolean_combo.append((factor, sympy.true))
+            else:
+                bool_cond = Or(*[Ne(cond.as_expr(), 0) for cond in conditions])
+                boolean_combo.append((factor, bool_cond))
+
+        results.append(boolean_combo)
+
+
+    conditions = None
+    if constant_conditions:
+        conditions = [Eq(cond.as_expr(), 0) for cond in constant_conditions]
```

Comment:
Since we are returning Poly here for the factors I think actually it would be better if this returns the conditions as Poly as well. The `factor_system_bool` function can turn these into booleans.

Reply:
I tried that, 

I have come up to something like:
```python
def factor_system_cond(eqs, gens=None, **kwargs):
    """Returns factors and conditions as Poly objects"""
    polys, result = _process_system_equations(eqs, gens, **kwargs)
    if result is not None:
        solutions, conditions = result
        if not solutions:
            return [], []
        if solutions == [[]]:
            return [[]], conditions if conditions else []
        return solutions, conditions if conditions else []

    poly_results, constant_conditions = factor_system_poly(polys, **kwargs)

    if not poly_results:
        if constant_conditions is None:
            return [], []
        return [], constant_conditions

    results = []
    for combo in poly_results:
        factor_conditions = {}
        for factor, conditions in combo:
            expr = factor.as_expr()
            if expr not in factor_conditions:
                factor_conditions[expr] = []
            factor_conditions[expr].extend(conditions)

        poly_combo = []  
        for factor, conditions in factor_conditions.items():
            if isinstance(factor, Expr):
                factor = Poly(factor, *polys[0].gens)
            if not conditions:
                poly_combo.append((factor, []))
            else:
                poly_combo.append((factor, conditions))

        results.append(poly_combo)

    return results, constant_conditions if constant_conditions else []
```

Am I catching your point correctly here ? It honestly confusing me a bit. 

For example:

Previously in our discussion above we were expecting an output from `factor_system_cond` like:
```python
>>> factor_system_cond([a * (x - 1), b], [x])
([[(x - 1, Ne(a, 0))]], [Eq(b, 0)])
```
Which is accurate and indicating the conditions clearly.

But the version mentioned in this comment behaves like:
```python
>>> factor_system_cond([a * (x - 1), b], [x])
([[(Poly(x - 1, x, domain='ZZ'), [Poly(a, x, domain='ZZ[a,b]')])]], [Poly(b, x, domain='ZZ[a,b]')])
```

But this seems kind of unclear to me, since here we can't really see what the conditions are supposed to be, it's not very clear to infer the conditions `Ne(a, 0)` and `Eq(b, 0)` from this output. 

 

> Since we are returning Poly here for the factors I think actually it would be better if this returns the conditions as Poly as well. The factor_system_bool function can turn these into booleans.

What am I missing here ? Please let me know, I might have taken it a bit off track here maybe.

Reply:
Yes that seems right to me. There isn't a way to represent inequalities with Poly so if we return Poly format then it should look as you've shown. The `factor_system_bool` function can be used to show exactly what all of the logical conditions mean.

Reply:
Would you suggest that we use `factor_system_bool` in `factor_system_cond` ? To present the logical conditions in an understandable format ?

Reply:
No, it should be the other way round: `factor_system_bool` calls `factor_system_cond` and converts the output from Poly back to Expr and to Equality/Unequality.